src/odinapi/views/level2.py

```diff
@@ -1144,10 +1160,23 @@ def _fetch(self, version, project):
             param = parse_parameters()
         except ValueError as e:
             raise BadRequest(str(e))
+
         db = level2db.Level2DB(project)
+        products = param['products']
         meas_iter = db.get_measurements(param.pop('products'), **param)
-        # TODO: Limit/paging
-        return meas_iter
+        if version == 'v4':
+            return meas_iter
+        scans = level2db.get_valid_collapsed_products(
+            list(meas_iter), get_args.get_int('document_limit'))
+        next_min_scanid = level2db.get_next_min_scanid(
+            list(meas_iter), get_args.get_int('document_limit'))
+        url = url_for(
+            'level2viewarea', project=project, version=version,
+            _external=True)
+        link = get_level2view_paging_link(
+            url, param, products, next_min_scanid)
+        headers = {'Link': link}
+        return scans, 200, headers
```

Comment:
Please use `HTTPStatus` rather than just numbers.

Reply:
That is a better convention, but that is not how it's done elsewhere in this file and having multiple conventions in the same file is not a good idea I think.

Reply:
Then one updates it a file at a time :wink:

Reply:
ok.